src/Config/Factory.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\ComposerDistributor\Config;
+
+use PharIo\ComposerDistributor\File;
+use PharIo\ComposerDistributor\FileList;
+use PharIo\ComposerDistributor\Url;
+
+class Factory
+{
+	/**
+	 * Config json example:
+	 *
+	 * {
+	 *   "packageName": "phar-io/phive",
+	 *   "keyDirectory: "keys",
+	 *   "phars": [
+	 *     {
+	 *       "name": "phive",
+	 *       "file:" "https://github.com/phar-io/phive/releases/%version%/phive.phar",
+	 *       "signature:" "https://github.com/phar-io/phive/releases/%version%/phive.phar.asc"
+	 *     }
+	 *   ]
+	 * }
+	 *
+	 */
+	public static function createFromFile(string $configFile): Config
```

Comment:
While I do understand the name of the method, technically it implies that the factory is createdFromString ;-)

Maybe we rename the thing into `Loader` and than can use `loadFile()`?

Reply:
I separated json loading and config creation in two separate classes `Config\Loader` and `Config\Factory` 
What do you think?

Reply:
Technically, the `Factory` is a `Mapper` then, but yes ;)

Reply:
So we keep it :)

Reply:
Well, a `Mapper` is good thing, not a bad thing.

Maybe it's a personal thing, but a `Factory` is something else for me. This thing merely translates an array into a DTO. That's exactly what mappers do ;)

Reply:
Rename it is ;)

Reply:
Done